packages/plugin-ext/src/main/browser/debug/plugin-debug-service.ts

```diff
@@ -159,19 +174,21 @@ export class PluginDebugService implements DebugService, PluginDebugAdapterContr
     async resolveDebugConfigurationWithSubstitutedVariables(config: DebugConfiguration, workspaceFolderUri: string | undefined): Promise<DebugConfiguration> {
         let resolved = config;
 
-        // we should iterate over all to handle configuration providers for `*`
-        for (const contributor of this.contributors.values()) {
-            if (contributor) {
-                try {
-                    const next = await contributor.resolveDebugConfigurationWithSubstitutedVariables(resolved, workspaceFolderUri);
-                    if (next) {
-                        resolved = next;
-                    } else {
-                        return resolved;
-                    }
-                } catch (e) {
-                    console.error(e);
+        const allProviders = Array.from(this.configurationProviders.values());
+        // Append debug type '*' at the end
+        const pluginProviders = allProviders.filter(p => p.type === config.type && !!p.resolveDebugConfigurationWithSubstitutedVariables);
+        pluginProviders.push(...allProviders.filter(p => p.type === '*' && !!p.resolveDebugConfigurationWithSubstitutedVariables));
```

Comment:
Alternative writing, feel free to ignore:

```suggestion
        const providersForType = allProviders.filter(p => p.type === config.type && !!p.resolveDebugConfigurationWithSubstitutedVariables);
        const providersForAny = allProviders.filter(p => p.type === '*' && !!p.resolveDebugConfigurationWithSubstitutedVariables));
        const pluginProviders = providersForType.concat(providersForAny);
```